lib/tasks/archive.rake

```diff
@@ -373,6 +376,13 @@ namespace :archive do
       puts "ERROR: could not find item pid=#{coll_id}-#{item_id} for file #{file} - skipping" if verbose
       return [basename, extension, coll_id, item_id, nil, nil]
     end
+
+    # don't allow too few or too many dashes
+    unless (remainder.count == 1 && remainder.none?(&:empty?)) || %w(CAT df PDSC_ADMIN).include?(remainder.last)
```

Comment:
This statement is fairly obtuse/unclear, maybe split it or explain it further in the comment.
I can't immediately tell why these cases are valid just by looking at it.